### PR TITLE
network/netdev: verify MAC address

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #21255.

### CODE REVIEW

File: src/network/netdev/veth.c [hunk 1]
```diff
@@ -58,20 +58,17 @@ static int netdev_veth_verify(NetDev *netdev, const char *filename) {
...
         if (!v->mac_peer) {
                 r = netdev_get_mac(v->ifname_peer, &v->mac_peer);
-                if (r < 0) {
-                        log_netdev_warning(netdev,
-                                           "Failed to generate predictable MAC address for %s. Ignoring",
-                                           v->ifname_peer);
-                        return -EINVAL;
-                }
+                if (r < 0)
+                        return log_netdev_warning_errno(netdev, r,
```
Comment (bluca) [inline src/network/netdev/veth.c]: `SYNTHETIC_ERRNO(EINVAL)`?

File: src/network/netdev/tunnel.c [hunk 2]
```diff
@@ -897,7 +900,7 @@ const NetDevVTable ip6tnl_vtable = {
         .fill_message_create = netdev_ip6tnl_fill_message_create,
         .create_type = NETDEV_CREATE_STACKED,
         .config_verify = netdev_tunnel_verify,
-        .generate_mac = true,
```
Comment (bluca) [inline src/network/netdev/tunnel.c:900]: it would be good to explain why `generate_mac` is being removed in the commit message

**Review by bluca — APPROVED** [review]

File: src/network/netdev/veth.c (hunk 1 above)
Comment (yuwata) [reply src/network/netdev/veth.c]: In this case, we should use real error cause. The commit message is updated.

File: src/network/netdev/tunnel.c (hunk 2 above)
Comment (yuwata) [reply src/network/netdev/tunnel.c:900]: Sure. Done.

**Comment by yuwata** [comment]
@bluca Thank you for your review. I updated the commit messages of the first two commits. Upgrading the green label.

**Pull request merged**